tutorial/lesson_22_serialization.cpp

```diff
@@ -0,0 +1,169 @@
+// Halide tutorial lesson 22: Serialization
+
+// This lesson describes how to serialize pipelines into a binary format 
+// which can be saved on disk, and later deserialized and loaded for 
+// evaluation.
+
+// Note that you'll need to be using a build of Halide that was configured
+// using the WITH_SERIALIZATION=ON macro defined in order for this tutorial
+// to work.
+
+// On linux, you can compile this tutorial and run it like so:
+// g++ lesson_22*.cpp -g -I <path/to/Halide.h> -L <path/to/libHalide.so> -lHalide -lpthread -ldl -o lesson_22 -std=c++17
+// LD_LIBRARY_PATH=<path/to/libHalide.so> ./lesson_22
+
+// On os x:
+// g++ lesson_22*.cpp -g -I <path/to/Halide.h> -L <path/to/libHalide.so> -lHalide -o lesson_22 -std=c++17
+// DYLD_LIBRARY_PATH=<path/to/libHalide.dylib> ./lesson_22
+
+// If you have the entire Halide source tree, you can also build it by
```

Comment:
...It makes me sad that our Tutorials are still relying entirely on Make, but that's a discussion for another day :-/

Reply:
They build with CMake too ...